### ISSUE TITLE
Rerouting application output to selected device

### ISSUE THREAD

**Issue by windwalker759** [naudio/NAudio#322]
Hi. I would like to reroute output of some applications to selected device. For example the output from game to headphones, output from browser to loudspeakers, etc. How I can realize that idea, and its possible with this library or not?

**Comment by markheath** [comment]
No, I'm afraid that's not possible with NAudio. You'd need a virtual audio driver to be able to do that.

**Comment by anmiles** [comment]
@markheath Isn't is possible in Windows 10? They added an interface to assign application to audio devices in Settings. I'd love to be able to do this programmatically using NAudio.
Isn't there a way to re-assign Session to another EndPoint?

**Comment by markheath** [comment]
It's not something I'm aware of, but it is of course quite possible that there are new WASAPI interfaces that support this. If there are, then perhaps we could create interop wrappers for them.

**Comment by jwallet** [comment]
maybe he wants something like [audiorouter](https://github.com/audiorouterdev/audio-router) but available as a nugget package

**Comment by markheath** [comment]
Looks a really interesting project - I've not seen it before. Maybe that could be done in C# - would need to dive into the implementation in more detail

**Comment by jwallet** [comment]
it works, I did use the solution from [SoundDeck](https://github.com/GeekyEggo/SoundDeck), https://github.com/jwallet/spy-spotify/issues/159

**Comment by markheath** [comment]
Closing this issue for now, as no plans to do anything in NAudio in this area at the moment. Thanks for the links @jwallet. We can reconsider in the future if more people ask for this / or a PR is submitted
